File: py/problem_3.py
# Problem 3
def primeFactors(n):
    p = 2
    while (n > p**2):
        if n % p == 0:
            print(p)
            n /= p
        else:
            p += 1
    
    print(n)

if __name__ == "__main__":
    print("The Prime Factors of 600851475143 are:")
    primeFactors(600851475143)

File: julia/problem_3.jl
# Problem 3

function primeFactors(n)
    p = 2
    factors = Int64[]
    while (n > p^2)
        if (n % p == 0)
            push!(factors, p)
            n = n / p
        else
            p += 1
        end
    end
    push!(factors, n)
    return factors

end

factors = primeFactors(600851475143)
println("Prime Factors of 600851475143 are $factors")